crates/ruff_linter/src/rules/pylint/rules/missing_maxsplit_arg.rs

```
@@ -29,14 +29,37 @@ use crate::checkers::ast::Checker;
 /// url = "www.example.com"
 /// prefix = url.split(".", maxsplit=1)[0]
 /// ```
+///
+/// To access the last element, use `str.rsplit()` instead of `str.split()`:
+/// ```python
+/// url = "www.example.com"
+/// suffix = url.rsplit(".", maxsplit=1)[-1]
+/// ```
 #[derive(ViolationMetadata)]
-pub(crate) struct MissingMaxsplitArg;
+pub(crate) struct MissingMaxsplitArg {
+    index: i64,
+    function: String,
+}
 
 impl Violation for MissingMaxsplitArg {
     #[derive_message_formats]
     fn message(&self) -> String {
-        "Accessing only the first or last element of `str.split()` without setting `maxsplit=1`"
-            .to_string()
+        let MissingMaxsplitArg { index, function } = self;
+
+        let correct_function = match index {
+            0 => "split",
+            -1 => "rsplit",
+            _ => "", // We should never hit this case
```

This should probably be an `unreachable!` instead, since if that case is hit it should cause a failure instead of a silent issue.

---

I think we should probably define a simple enum instead of using an `i64`. That way we can check for known values up front and bail out of the whole diagnostic. Then we can `match` exhaustively here.